### ISSUE TITLE
Submodules showing branch name of parent project

### ISSUE THREAD

**Issue by timo-rohwedder** [zielu/GitToolBox#328]
Hi!

I'm not sure if this is a bug, a misconfiguration or something special in my individual setup.
This happens since I updated to IntelliJ 2020.2. I am using the latest EAP Version of GitToolBox and also tried to downgrade to the latest stable version.

I have an aggregator pom as an intellij-module in my project and also all the modules inside the aggregator pom.

Some (the most) of the submodules now showing the current branch name of the parent module next to their name.

Any idea? Thanks for your great work!

**Comment by zielu** [comment]
Hi, this falls into category of known issue. Logic is very simple there: if something is a module it shows branch. For multimodule projects this is confusing. I'll take a look if it can be hidden when module is child of another module.

**Comment by gmarjano1975** [comment]
Hi, I got the same problem - but even weirder. I got a parent Module and three submodules. One submodule A changes the branch name in the project view - the two other submodules (B,C) don't. They always show the parent branch name.

I compared the pomfiles of A and B,C and they are the same regarding the <parent>-Tag.

Very confusing. At the moment I must disable the plugin to avoid being on the wrong branch.

**Comment by zielu** [comment]
@gmarjano1975 you are probably referring to git submodules. Here issue is with one single repo containing many maven modules.
Git submodules are not supported yet

**Comment by zielu** [comment]
#330 looks related

**Comment by timo-rohwedder** [comment]
Duplicate of #330

**Comment by zielu** [comment]
Let's see if original issue goes away after #330. If not I'll reopen this one.

**Comment by zielu** [comment]
Reopening this one #330 was a different issue

**Comment by jrmihalick** [comment]
@zielu I have a gradle project that is using submodules. Prior to updating to  to version 2021.x, I used to see the branch name decoration on all of my submodules.  Now it just shows on one of any of the given submodules and it actually moves around to different submodules after I pull updates from git.  Is there any way to get the decoration back on all my projects?
![Screen Shot 2022-01-20 at 11 52 17 AM](https://user-images.githubusercontent.com/11838640/150385246-9695c60c-3fde-4079-99a7-07bb5f010f63.png)

**Comment by zielu** [comment]
@jrmihalick this is side effect of adding logic to only show branches on top level repo instead of all module tree nodes. I forgot that there could be git submodules too. I'll see what I can do.

**Comment by jrmihalick** [comment]
@zielu Thanks.  If you could make it an option that we can toggle on/off, our team of 30 or so developers would really appreciate it.

**Comment by zielu** [comment]
Looks like I need some more details to reproduce it. Is the project structured like this ?
- repo
  - submodule1
  - submodule2
 
Are submodule repos added as maven/gradle modules to root repo ?

**Comment by jrmihalick** [comment]
@zielu The project is structured like so:

- A single git repository, not using git submodules
- There is a gradle parent project, which adds the sibling subproject dirs via gradle `include` mechanism.

Structure:

```
container-project-dir (directory, root of git repository)
  parent-assemble (directory, subdir in git repo)
  child-project-1 (directory, subdir in git repo)
  child-project-2 (directory, subdir in git repo)
  ...
  child-project-n (directory, subdir in git repo)
```

In the parent-assemble project, we have a `settings.gradle` which uses the gradle include mechanism to add the child projects like so (see https://docs.gradle.org/current/userguide/multi_project_builds.html):

```
    include 'child-project-1'
    include 'child-project-2'
    ...
    include 'child-project-n'
```

This results in an intellij project structure like so:
```
  parent-assemble
  child-project-1
  child-project-2
  ...
  child-project-n
```

You can view an example project of ours here: https://github.com/JumpMind/openpos-framework

- The git repo is `openpos-framework`.
- The parent-assemble project is `openpos-assemble`.
- The child projects are the siblings to `openpos-assemble`, which are `openpos-client-libs`, `openpos-client-personalize`, `openpos-devices`, etc.  
- You can see in my original screenshot above how these projects are laid out in the resulting Intellij project.

**Comment by zielu** [comment]
@jrmihalick which directory do you open as project in Idea ? `openpos-assemble` or `openpos-framework` ?

**Comment by jrmihalick** [comment]
@zielu The gradle `idea` plugin generates an ipr file into `openpos-assemble`, so we can open the `openpos-assemble.ipr` file from there.  We don't point intellij at a directory per se.

We have other projects where the ipr file is generated and placed in the parent directory of the parent-assemble project and can have an arbitrary name and just organizes all the projects to be viewed together as siblings in intellij.  For example, we have intellij projects where the child projects come from 3 different git repositories, but they all appear as siblings to each other in the Intellij project workspace.  Intellij figures this out, and your plugin used to also correctly figure it out and show the correct repository decorations next to each and every child project.

**Comment by zielu** [comment]
@jrmihalick I reproduced it - fix is on the way

**Comment by jrmihalick** [comment]
@zielu : Great news!  Thanks!

**Comment by zielu** [comment]
@jrmihalick EAP build x.8.6 should be out in couple of minutes - see https://github.com/zielu/GitToolBox#eap-builds how to switch to EAP channel

**Comment by zielu** [comment]
Done in variant.8.8
